Assets/MRTK/StandardAssets/EditorUtilities/OnLoadUtilities.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
+{
+    [InitializeOnLoad]
+    public static class OnLoadUtilities
+    {
+        private const string ShaderSentinelFile = "MRTK.Shaders.Sentinel";
+        private const string ShaderImportDestination = "MRTK/Shaders";
+
+        static OnLoadUtilities()
+        {
+            ImportShaderFiles();
+            EnsureShaders();
+        }
+
+        /// <summary>
+        /// Ensures that MRTK shader files are present in the Assets folder tree as they
+        /// need to be modifiable to support the Universal Render Pipeline
+        /// </summary>
+        private static void EnsureShaders()
+        {
+            if (!AssetsContainsShaders())
+            {
+                ImportShaderFiles();
+            }
+        }
+
+        /// <summary>
+        /// Checks to see if the Assets folder tree contains the MRTK shaders.
+        /// </summary>
+        /// <returns>True if the shader sentinel file is found, otherwise false.</returns>
+        private static bool AssetsContainsShaders()
+        {
+            string assetRoot = Application.dataPath;
+            DirectoryInfo di = new DirectoryInfo(assetRoot);
+            return (di.GetFiles(ShaderSentinelFile, SearchOption.AllDirectories).Length > 0);
+        }
+
+        /// <summary>
+        /// Finds the shader folder within the package cache.
+        /// </summary>
+        /// <returns>
+        /// DirectoryInfo object representing the shader folder in the package cache.
+        /// If not found, returns null.</returns>
+        private static DirectoryInfo FindShaderFolderInPackage()
+        {
+            DirectoryInfo di = new DirectoryInfo(Path.GetFullPath(Path.Combine("Library", "PackageCache")));
```

Comment:
Can we search for this via GUID or something instead? The more we assume things are in `Library/PackageCache`, the less possible it is for users to use MRTK as an [embedded](https://docs.unity3d.com/Manual/upm-embed.html) (which is easier to solve, if we check `Packages` as well like [MRTKFiles](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/31ff4c9111b4a2c9bcda58e5db2136a44eca65f8/Assets/MRTK/Core/Utilities/Editor/Setup/MixedRealityToolkitFiles.cs#L256) does) or [local](https://docs.unity3d.com/Manual/upm-localpath.html) package (and the harder it'll be for us to enable that if it's blocked!)

Reply:
Urg but I guess GUID might not be visible since this is in a `~` folder...

Reply:
that is why i took this route.

Reply:
Now that `AssetsContainsShaders` is updated for embedded packages. I wanted to revive this conversation and my suggestion above

>for users to use MRTK as an [embedded](https://docs.unity3d.com/Manual/upm-embed.html) (which is easier to solve, if we check `Packages` as well like [MRTKFiles](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/31ff4c9111b4a2c9bcda58e5db2136a44eca65f8/Assets/MRTK/Core/Utilities/Editor/Setup/MixedRealityToolkitFiles.cs#L256) does)

Should we add `Packages` as a search location here?

Reply:
good thought. the editor may not be able to find the file since unpacked, it will be in a ~ folder

Reply:
added support for embedded package having a Shaders~ folder instead of Shaders